magnet/predictor.py

```diff
@@ -248,6 +156,124 @@ def _coerce_helm_suite_inputs(self, *args, **kwargs):
             helm_suite_path = ub.Path(args[0])
         return helm_suite_path
 
+    def _run(self):
+        raise NotImplementedError
+
+    def __call__(self, *args, **kwargs):
+        return self._run(*args, **kwargs)
+
+
+class RunPrediction:
+    def __init__(self,
+                 run_spec_name,
+                 split,
+                 stat_name,
+                 mean,
+                 computed_on=None,
+                 perturbation_parameters=None,
+                 count=1,
+                 sum=None,
+                 sum_squared=None,
+                 min=None,
+                 max=None,
+                 variance=0.0,
+                 stddev=0.0):
+        self.run_spec_name = run_spec_name
+        self.split = split
+        self.stat_name = stat_name
+        self.mean = mean
+
+        self.computed_on = computed_on
+
+        if perturbation_parameters is None:
+            perturbation_parameters = {}
+        self.perturbation_parameters = {
+               **{"name": None, "fairness": None, "robustness": None},
+               **perturbation_parameters}
+
+        self.count = count
+
+        self.sum = sum
+        if self.sum is None:
+            self.sum = mean
```

Comment:
Are you sure you want this to be the mean? If anything `mean * count` seems correct. Is there a reason to even check for none values here? It will slow down the creation of instances, and it also obfuscates if the values were null to begin with.

Maybe just do the simple thing in the `__init__`  by setting the instance attributes (maybe even implicitly using a dataclass), and if you want you could add a `_populate` method that infers missing values when they are inferrable. I'm not a big fan of putting incorrect values in things like `max` when we don't know what the max actually was. But things like sum squared make some sense when we have sum.

Still, unless there is a reason to postprocess, I'd probably just pass the nones through as-is.

Reply:
Hmm, you're right there `mean * count` would be more correct.  And I think you're right that it's probably better to just leave `None` values as such, also because we don't merge on or display those fields anyway right now.  Let me tweak some stuff there